### ISSUE TITLE
Use spotless plugin to format the code to ensure that the format of the submitted code is uniform

### ISSUE THREAD

**Issue by liugddx** [alibaba/nacos#11021]
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by liugddx** [comment]
@KomachiSion  Can we format code with spotless?

**Comment by KomachiSion** [comment]
No, maven plugin reformat code will cause many invalid the commits in PR. We prefer use codeStyle to do it when developer developing.

**Comment by liugddx** [comment]
> No, maven plugin reformat code will cause many invalid the commits in PR. We prefer use codeStyle to do it when developer developing.

emm.. you're right. This is a case used by apache shardingsphere, maybe can refer to avoid invalid code changes

https://medium.com/codex/how-does-apache-shardingsphere-standardize-and-format-code-we-use-spotless-caf09403de6a

**Comment by KomachiSion** [comment]
Whether the link is right? I can't open it even i use vpn.  

If use maven plugin, whether the two problem solved?

1. developers do mvn build command in local and just do the commit in PR, it will has one invalid commit about the code reformat. It can't avoid the review in code when developer **not** do mvn build command.
2. if in work flow add the check for reformat plugin, it will cause the ci flow has some modifies need to commit.

I know the sofa use this way to auto check the reformat way, But We prefer developer can use codeStyle to do reformat during developing.

Of cource, if the two problem can be solved well, we also welcome use plugin to auto reformat.

**Comment by liugddx** [comment]
Thank you for your reply. Both points you mentioned can be satisfied.I can submit a pr to fix it.

**Comment by liugddx** [comment]
https://nightlies.apache.org/flink/flink-docs-stable/docs/flinkdev/ide_setup/#intellij-idea

This is the formatting of the code of Apache Flink.We can refer to it.

**Comment by KomachiSion** [comment]
The document in nacos also included, both document and source codes.

From your PR, I can't get the key point. Did you means that the spotless plugin conflict codestyle plugin?

**Comment by liugddx** [comment]
> The document in nacos also included, both document and source codes.
> 
> From your PR, I can't get the key point. Did you means that the spotless plugin conflict codestyle plugin?

Yes, when I run `mvn spotless:apply`, the code will be formatted and checkstyle will verify that these changes do not comply with. This is caused by the inconsistent rules of the two plug-ins.

**Comment by KomachiSion** [comment]
so should we use double maven commands? for example: first run spotless without checkstyle and then run check command in action.

**Comment by liugddx** [comment]
https://github.com/apache/doris/pull/25033 

My pr in apache doris, but I removed the checkstyle check. I can't resolve the conflict between these two plugins, therefore, I chose to use spotless to cover the checkstyle rules as much as possible.

**Comment by KomachiSion** [comment]
So use checkstyle to keep running first.  If spotless can replace checkstyle fully, then use spotless replace it.

**Comment by miclosgyjf** [comment]
> **¿Su solicitud de función está relacionada con un problema? Por favor describa.** Una descripción clara y concisa de cuál es el problema. Ex. Siempre me siento frustrado cuando [...]
> 
> **Describe la solución que te gustaría.** Una descripción clara y concisa de lo que quieres que suceda.
> 
> **Describa las alternativas que haya considerado.** Una descripción clara y concisa de cualquier solución o característica alternativa que haya considerado.
> 
> **Contexto adicional** Agregue aquí cualquier otro contexto o captura de pantalla sobre la solicitud de función.

https://github.com/alibaba/nacos/issues/11021#issue-1866759696
